File: semester_08/Desarrollo_Movil/Actividad7/app/src/main/java/com/example/actividad7/MainActivity.java
package com.example.actividad7;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;
public class MainActivity extends Activity {
    private Button b;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        b = (Button) findViewById(R.id.button);
        b.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Toast.makeText(MainActivity.this, "Button 2 ", 2500).show();
            }
        } );
    }
}

File: semester_08/Desarrollo_Movil/Actividad26/app/src/main/java/com/example/actividad26/MainActivity.kt
package com.example.actividad26


import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import androidx.activity.ComponentActivity
import com.example.actividad26.db.MyDbHelper
import com.example.actividad26.db.ToDoTable


class MainActivity : ComponentActivity() {
    private val todos = ArrayList<ToDo>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val todoAdapter = ArrayAdapter<ToDo>(
            this,
            android.R.layout.simple_list_item_1,
            android.R.id.text1,
            todos
        )
        val db=MyDbHelper(this).writableDatabase
        fun refreshTodoList(){
            val todoList=ToDoTable.getAllNodos(db)
            // Log.d("TODOS",todoList.toString())
            todos.clear()
            todos.addAll(todoList)
            todoAdapter.notifyDataSetChanged()
        }
        refreshTodoList()
        //set todoadapter in listView
        val etNewToDo = findViewById<EditText>(R.id.etNewToDo)
        val btnAddToDo = findViewById<Button>(R.id.btnAddToDo)
        val lvTools = findViewById<ListView>(R.id.lvTools)
        lvTools.adapter = todoAdapter
        btnAddToDo.setOnClickListener {
            val newToDo=ToDo(
                etNewToDo.text.toString(),
                false
            )
            ToDoTable.insertTodo(db,newToDo)
            refreshTodoList()
        }
    }
}